File: algorithms/problem-solving/class-photo/class.ts
export function classPhotos(redShirtHeights: number[], blueShirtHeights: number[]): boolean {
  redShirtHeights.sort()
  blueShirtHeights.sort()
  const backRow =
    redShirtHeights[redShirtHeights.length - 1] > blueShirtHeights[blueShirtHeights.length - 1]
      ? redShirtHeights
      : blueShirtHeights
  const frontRow =
    redShirtHeights[redShirtHeights.length - 1] < blueShirtHeights[blueShirtHeights.length - 1]
      ? redShirtHeights
      : blueShirtHeights

  let c = 0
  for (let i = 0; i < backRow.length; i++) {
    if (backRow[i] > frontRow[i]) {
      c++
    }
  }

  return c === frontRow.length
}

// const r = classPhotos(reds, blues)

File: problem-solving/class-photo/class.go
package main

import (
	"fmt"
	"sort"
)

// ClassPhotos func
func ClassPhotos(redShirtHeights []int, blueShirtHeights []int) bool {

	sort.Ints(redShirtHeights)
	sort.Ints(blueShirtHeights)

	front:= []int{}
	back:= []int{}

	if redShirtHeights[len(redShirtHeights)-1] > blueShirtHeights[len(blueShirtHeights)-1] {
		front = redShirtHeights
	}else{
		front = blueShirtHeights
	}

	if redShirtHeights[len(redShirtHeights)-1] < blueShirtHeights[len(blueShirtHeights)-1] {
		back = redShirtHeights
	}else{
		back = blueShirtHeights
	}

	count:=0
	for i := range front {
		if front[i] > back[i] {
			count++
		}
	}

	return count == len(front)
}


func main(){
	 reds := []int{5, 8, 1, 3, 4}
 	blues := []int{6, 9, 2, 4, 5}
	res:= ClassPhotos(reds,blues)
	fmt.Println(res)
}